k_tree/find_matches.py

```diff
@@ -0,0 +1,20 @@
+"""Return the nodes that match the val."""
+
+from .k_tree import KTree
+
+
+def find_matches(tree, val):
+    """Return the nodes that match the val."""
+    new = KTree()
+
+    def find(node):
+        nonlocal new, val
```

Comment:
You do not need to set these as `nonlocal` because you are not reassigning the variable at any point.